ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -95,24 +104,68 @@ const WalletPaymentPickMethodScreen = () => {
   // only when the transaction status becomes ACTIVATED.
   useOnTransactionActivationEffect(
     React.useCallback(() => {
-      navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
-        screen: WalletPaymentRoutes.WALLET_PAYMENT_PICK_PSP
-      });
+      pipe(
+        sequenceT(O.Monad)(
+          pot.toOption(paymentAmountPot),
+          pot.toOption(transactionPot),
+          selectedWalletOption
+        ),
+        O.map(([paymentAmountInCents, transaction, selectedWallet]) => {
+          const transferList = transaction.payments.reduce(
+            (a, p) => [...a, ...(p.transferList ?? [])],
+            [] as ReadonlyArray<Transfer>
+          );
+          const paymentToken = transaction.payments[0]?.paymentToken;
+
+          dispatch(
+            walletPaymentCalculateFees.request({
+              paymentToken,
+              paymentMethodId: selectedWallet.paymentMethodId,
+              walletId: selectedWallet.walletId,
+              paymentAmount: paymentAmountInCents,
+              transferList
+            })
+          );
+        })
+      );
       setWaitingTransactionActivation(false);
-    }, [navigation])
+    }, [dispatch, paymentAmountPot, selectedWalletOption, transactionPot])
   );
 
+  React.useEffect(() => {
+    if (!pot.isLoading(pspListPot) && !pot.isError(pspListPot)) {
+      pipe(
+        pspListPot,
+        pot.toOption,
+        O.map(pspList => {
+          if (pspList.length > 1) {
+            navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
+              screen: WalletPaymentRoutes.WALLET_PAYMENT_PICK_PSP
+            });
+          } else if (pspList.length >= 1) {
+            navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
+              screen: WalletPaymentRoutes.WALLET_PAYMENT_CONFIRM
+            });
+          }
+        })
+      );
+    }
+  }, [navigation, pspListPot]);
```

Comment:
The `pot.toOption` function already covers the conditions inside the `if` statement, so it could be removed. What do you think?

```suggestion
      pipe(
        pspListPot,
        pot.toOption,
        O.map(pspList => {
          if (pspList.length > 1) {
            navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
              screen: WalletPaymentRoutes.WALLET_PAYMENT_PICK_PSP
            });
          } else if (pspList.length >= 1) {
            navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
              screen: WalletPaymentRoutes.WALLET_PAYMENT_CONFIRM
            });
          }
        })
      );
  }, [navigation, pspListPot]);
```

Reply:
That's useful! Thanks for the suggestion, I addressed it into b0a4f3127501b168bf60dddc3bffdd0ece0b4f6e